Extensions/v0.7 extensions (#86)
kwilteam/docs: docs/extensions/utility-functions.mdx

---
sidebar_position: 6
sidebar_label: "Utilities"
id: extension-utils
title: "Utilities"
description: Utilities for working with extensions
slug: /extensions/utils
---

> Kwil provides utilities for altering core network state from within extensions. With a database connection, you can use these utilities to get user balances,

Reviewer: What is the package(s) that expose the utility functions? Not clear in these docs.

> Kwil provides utilities for altering core network state from within extensions. With a database connection, you can use these utilities to get user balances,
> transfer tokens, update validator power, and more. These can be accessed in the "github.com/kwilteam/kwil-db/common/functions" package. For example, to credit
> an Ethereum address with 100 tokens, you can use the `Credit` function.

Reviewer: ```suggestion
Kwil provides utilities for altering core network state from within extensions. With a database connection, you can use these utilities to get user balances,
transfer tokens, update validator power, and more. These are accessed in the [`github.com/kwilteam/kwil-db/common/functions`](https://pkg.go.dev/github.com/kwilteam/kwil-db/common/functions) package. For example, to credit
an Ethereum address with 100 tokens, you can use the `Credit` function.
```

```go
package main

import (
	"context"
	"math/big"

	"github.com/ethereum/go-ethereum/common"
	"github.com/kwilteam/kwil-db/common/functions"

> 	"github.com/kwilteam/kwil-db/common/sql"

Reviewer: why are you importing `common/sql`?

Author: function signature uses sql.DB
